src/main/java/com/socialmedia/service/UserService.java

```diff
@@ -104,10 +104,11 @@ public Boolean confirmEmail(String confirmationId) {
   public ApplicationUser delete(String id) {
     ApplicationUser deletedUser = resolvedOptional(jpaRepository.findById(id), id);
 
-    deletedUser.getFriends().forEach(friend->removeFriend(friend, id));
+    deletedUser.getFriends().forEach(friend -> removeFriend(friend, id));
     deletedUser.getChats().forEach(chat-> chatService.removeParticipant(chat, id));
-    friendRequestService.getAllByRequester(deletedUser).forEach(request -> friendRequestService.delete(request.getId()));
-    friendRequestService.getAllByResponder(deletedUser).forEach(request -> friendRequestService.delete(request.getId()));
+    friendRequestService.getAllByRequester(deletedUser).forEach(request-> friendRequestService.delete(request.getId()));
+    friendRequestService.getAllByResponder(deletedUser).forEach(request-> friendRequestService.delete(request.getId()));
+    postService.findAllUsersPosts().forEach(post -> postService.delete(post.getId()));
```

Comment:
Currently findAllUsersPosts return posts where deleted user was owner. But you also need to delete posts that deleted user authored but not owning

Reply:
Agree